File: examples/java/kora-java-resilient/src/main/java/ru/tinkoff/kora/example/resilient/CircuitBreakerService.java
package ru.tinkoff.kora.example.resilient;

import ru.tinkoff.kora.common.Component;
import ru.tinkoff.kora.resilient.circuitbreaker.annotation.CircuitBreaker;

@Component
public class CircuitBreakerService {

    @CircuitBreaker("my_cb")
    public String getSuccessful() {
        return "OK";
    }
}

File: examples/kotlin/kora-kotlin-resilient/src/main/kotlin/io/koraframework/kotlin/example/resilient/CircuitBreakerService.kt
package io.koraframework.kotlin.example.resilient

import io.koraframework.common.annotation.Component
import io.koraframework.common.annotation.Root
import io.koraframework.resilient.circuitbreaker.annotation.CircuitBreaker
import io.koraframework.resilient.fallback.annotation.Fallback
import io.koraframework.resilient.retry.annotation.Retry
import io.koraframework.resilient.timeout.annotation.Timeout
import java.util.concurrent.ThreadLocalRandom

@Component
open class CircuitBreakerService {
    @CircuitBreaker("my_cb")
    open fun getSuccessful(): String = "OK"
}

